app/models/certificate.rb

```diff
@@ -0,0 +1,29 @@
+class Certificate < ApplicationRecord
+  include WithCode
+
+  belongs_to :user
+  belongs_to :certification
+
+  has_one :organization, through: :certification
+
+  delegate :title, :description, :template_html_erb, :background_image_url, to: :certification
+
+  def self.code_size
+    12
+  end
+
+  def filename
+    "#{title.parameterize.underscore}.pdf"
+  end
+
+  def locals
+    json = as_json only: [:start_date, :end_date],
+                   include: {
+                     user: { methods: [:formal_first_name, :formal_last_name, :formal_full_name] },
+                     certification: { only: [:title, :description] },
+                     organization: { only: [:name, :display_name] }
+                   }
+    JSON.parse json.to_json, object_class: OpenStruct
```

Comment:
Maybe we can extract this behavior to a method in Hash, like `to_os`?

Reply:
This methods already exists. It is `to_struct`: https://github.com/mumuki/mumukit-core/blob/2f822e552f4f284b8a3563283e4b0c00a409d840/lib/mumukit/core/hash.rb#L33.


The new thing here is that conversion is deep, right? In that case we could have `to_deep_struct`. Anyway, I don't recommend using intermediate JSON generation and parsing, because it is unnecessary slow

Reply:
Yes, that's true! `to_deep_struct` seems like a good name!
We could convert the hash recursively instead of doing this. I think this was the more elegant solution but not the fastest one :thinking:

Reply:
![image](https://user-images.githubusercontent.com/11860076/108872497-ff354900-75d8-11eb-9e9d-e030e7d05249.png)
There is a big difference, so let's please use a custom solution for this one